### ISSUE TITLE
Java Access Bridge: 64-bit architecture support

### ISSUE THREAD

**Issue by OzancanKaratas** [nvaccess/nvda#7609]
In 64-bit based systems, 64-bit Java installation is not recognized by NVDA so 64-bit Java Access Bridge doesn't works.

The 32-bit Java is running normally, but 64-bit support should also be added.

**Comment by Brian1Gaff** [comment]
Don't you have to enable them separately to make them both work?
 Its been a while since I needed them.
 Brian

bglists@blueyonder.co.uk 
Sent via blueyonder. 
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by OzancanKaratas** [comment]
If both there enabled, will be available that latest installed Java installation's Access Bridge.
So I installed both 32-bit Java and 64-bit Java, but we should support 64-bit architecture.

If we do not support it, 64-bit Java is not used no way.

**Comment by Brian1Gaff** [comment]
I'm not sure what you are saying here. I distinctly recall having to turn on 
access bridge using a command line switch for both 64 and 32 bit versions. 
As I say I do not have either on here any more as most Java software seems 
sluggish to me on even fast machines so I ababandoned the thing.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by josephsl** [comment]
Hi, isn’t this a duplicate of several issues regarding JAB64? Thanks.

**Comment by OzancanKaratas** [comment]
If so then I apologise. I searched, but I didn't found an issue about this.

**Comment by leonardder** [comment]
Since this issue is suggesting Adding Java access Bridge X64 support to NVDA (which is already there), I'm closing this as invalid. If the X64 Access Bridge doesn't work properly on your end and you are pretty sure that this is NVDA's fault, please open a new ticket or search for an existing one.

When I install Java, I Always doe it with [ninite](http://ninite.com/], which installs both x86 and x64 versions of the JRE. When I activate the access bridge in the Ease of access center>signin options (from the legacy control panel), both X86 and X64 bridges do their job pretty well.

**Comment by Brian1Gaff** [comment]
Don't know the issue search on github is obviously designed by somebody with 
a sadistic streak or at least does not use the same terms we would use to 
find things!:-)

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by OzancanKaratas** [comment]
Hi @leonardder, I'm sure that Java Access Bridge's 64-bit version do not works with NVDA, this problem causes by NVDA.

Please reopen this issue.

**Comment by josephsl** [comment]
Hi, 64-bit Java apps are not supported properly because NVDA is a 32-bit application. I’m not really willing to reopen this as this is a duplicate of several issues. Thanks.

**Comment by OzancanKaratas** [comment]
But 64-bit SAPI and OneCore is supported, this does not make sense.

**Comment by josephsl** [comment]
Hi, according to the source code, the name of the bridge DLL is hardcoded, meaning that NVDA can’t really use 64-bit JAB unless there’s a way to dynamically load one or both on the fly. SAPI and OneCore support was possible because Windows provides a way to find out which voices are available and a way to retrieve the correct DLL via Windows Registry. Thanks.

**Comment by OzancanKaratas** [comment]
Alright, thanks. I'll contact to Oracle. They will help us out.

**Comment by leonardder** [comment]
It is a real misunderstanding that 64-bit JAVA applications are not supported, as I'm able to use 64-bit JAVA applications just fine. For example, I've been able to use IBM SPSS Statistics X64 properly. See #2567 and #1915.
